python/ray/tests/test_chaos.py

```diff
@@ -123,6 +127,92 @@ def get(self):
     # assert_no_system_failure(p, 10)
 
 
+@pytest.fixture
+def set_kill_interval(request):
+    lineage_reconstruction_enabled, kill_interval = request.param
+    os.environ["RAY_lineage_pinning_enabled"] = ("1" if
+                                                 lineage_reconstruction_enabled
+                                                 else "0")
+    request.param = {
+        "kill_interval": kill_interval,
+        "head_resources": {
+            "CPU": 1
+        },
+        "worker_node_types": {
+            "cpu_node": {
+                "resources": {
+                    "CPU": 2,
+                },
+                "node_config": {},
+                "min_workers": 0,
+                "max_workers": 4,
+            },
+        },
+        # We just want the autoscaler to restart nodes, not take them down.
+        "idle_timeout_minutes": 10,
+    }
+    cluster = _ray_start_chaos_cluster(request)
+    while True:
+        try:
+            yield (lineage_reconstruction_enabled, kill_interval,
+                   next(cluster))
+        except StopIteration:
+            break
+    del os.environ["RAY_lineage_pinning_enabled"]
+
+
+@pytest.mark.parametrize(
+    "set_kill_interval", [(True, None), (True, 30), (False, None),
+                          (False, 30)],
```

Comment:
Isn't 30 seconds interval too long? How many nodes are usually crashed per test?

Reply:
I think this is about right...the test takes at least a minute on my laptop, so this will kill on average 2 nodes/test. If we kill too often, then lineage reconstruction can "hang" because it doesn't have enough resources to finish.

Reply:
Makes sense. Maybe we need some sort of exit case here, but practically it is probably not that common in real world